src/main/java/duke/dukeobject/ExpenseList.java

```diff
@@ -117,46 +123,42 @@ public void setViewScope(String viewScope) {
     /**
      * Tailors the given List so that only {@code Expense} within the given time scope are preserved.
      * Returns the tailored List.
+     *
      * @param currentList The list going to be modified.
      * @return The tailored List.
      */
     @Override
     public List<Expense> view(List<Expense> currentList) {
         List<Expense> viewedList = new ArrayList<Expense>();
         LocalDateTime now = LocalDateTime.now();
+
         switch(viewScope) {
-        case "all":
-            viewedList = currentList;
-            break;
-        case "year":
-            for(Expense e : currentList) {
-                if(e.getTime().plusYears(1).isAfter(now)) {
-                    viewedList.add(e);
-                }
-            }
-            break;
-        case "month":
-            for(Expense e : currentList) {
-                if(e.getTime().plusMonths(1).isAfter(now)) {
-                    viewedList.add(e);
-                }
-            }
-            break;
-        case "week":
-            for(Expense e : currentList) {
-                if(e.getTime().plusDays(7).isAfter(now)) {
-                    viewedList.add(e);
-                }
-            }
-            break;
-        case "day":
-            for(Expense e : currentList) {
-                if(e.getTime().plusDays(1).isAfter(now)) {
-                    viewedList.add(e);
-                }
-            }
-        default:
-            throw new DukeRuntimeException("Incorrect format of view scope");
+            case ALL:
+                viewedList = currentList;
+                break;
+
+            case YEAR:
+                viewedList = currentList.stream()
```

Comment:
for all of these, the returned list should be expenses in the same year/month/week/day rather rhan within a certain range of the current day.